packages/jazz-svelte/src/lib/tests/components/CoState/UpdateNestedValue.svelte

```diff
@@ -0,0 +1,36 @@
+<script lang="ts" module>
+  export type Props = {
+    id: ID<Person>;
+  };
+</script>
+
+<script lang="ts">
+  import { CoState } from '../../../jazz.class.svelte.js';
+
+  import { type ID } from 'jazz-tools';
+  import { Person } from './schema.js';
+
+  let props: Props = $props();
+
+  const person = $derived(
+    new CoState(Person, props.id, {
+      resolve: {
+        dog: true
+      }
+    })
+  );
```

Comment:
Had to wrap this with `$derived` to make it react to id changes.

Tried with `() => props.id` but didn't work.

Will make a separate PR for reactive ids